### ISSUE TITLE
The `AtomInteger` trait is not needed for `Atomic::fetch_update`

### ISSUE THREAD

**Issue by bruceg** [LukasKalbertodt/atomig#3]
The function `Atomic::fetch_update` is only available for types that implement `AtomInteger`. However, it doesn't actually do any implicit integer operations like, say, `fn fetch_add` does — it *explicitly* leaves the update to a Rust closure that is restricted to safe operations on the data types. As such, the restriction is not needed, allowing `fetch_update` to be used on `enum` types, for example.

Here's my example that does not work with atomig as it is now:

```rust
#[derive(Atom, Copy)]
#[repr(u8)]
enum Enum {
    Off,
    On,
}

fn foo() {
    let t: Atomic<Enum> = Atomic::new(Enum::Off);
    t.fetch_update(Ordering::Relaxed, Ordering::Relaxed, |t0| Some(match t0 {
        Enum::Off => Enum::On,
        Enum::On => Enum::Off,
    }));
}
```

**Comment by LukasKalbertodt** [comment]
Somehow I managed to unwatch my own repository :astonished: I just now stumbled upon your comment by pure chance. Sorry for the delay!

The problem here is that `fetch_update` was not implemented for non-integer atomics in std for some time: https://github.com/rust-lang/rust/issues/78639
This seems to be a simple oversight though. `AtomicBool::fetch_update` and `AtomicPtr::fetch_update` were already added and their stabilization hits stable with 1.53, in roughly three weeks. I will wait till then to properly fix this issue. I will let you know once I release the fix as 0.3.
